test/unit/slc/utest_slc.py

```diff
@@ -0,0 +1,219 @@
+import contextlib
+from pathlib import Path
+from test.unit.slc.util import (
+    SecretsMock,
+    not_raises,
+)
+from unittest.mock import (
+    Mock,
+    create_autospec,
+)
+
+import git
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+
+from exasol.nb_connector.secret_store import Secrets
+from exasol.nb_connector.slc import (
+    constants,
+    script_language_container,
+    workspace,
+)
+from exasol.nb_connector.slc.script_language_container import ScriptLanguageContainer
+from exasol.nb_connector.slc.slc_flavor import (
+    SlcError,
+    SlcFlavor,
+)
+from exasol.nb_connector.slc.workspace import current_directory
+
+
+@pytest.fixture
+def sample_slc_name() -> str:
+    return "CUDA"
+
+
+@pytest.fixture
+def git_repo_mock(monkeypatch: MonkeyPatch):
+    mock = create_autospec(git.Repo)
+    monkeypatch.setattr(workspace, "Repo", mock)
+    return mock
+
+
+def simulate_clone(flavor: str):
+    """
+    This function can be set as side effect to method clone_from of the
+    git_repo_mock for simulating a successful checkout to the
+    ScriptLanguageContainer.
+    """
+
+    def create_dir(url: str, dir: Path, branch: str):
+        (dir / constants.FLAVORS_PATH_IN_SLC_REPO / flavor).mkdir(parents=True)
+        return Mock()
+
+    return create_dir
+
+
+@pytest.fixture
+def slc_factory(tmp_path, git_repo_mock):
+    """
+    Provides a context to create an instance of ScriptLanguageContainer in
+    a temporary directory and simulating the SLC Git repo to be cloned inside.
+    """
+
+    @contextlib.contextmanager
+    def context(slc_name: str, flavor: str):
+        secrets = SecretsMock(slc_name)
+        git_repo_mock.clone_from.side_effect = simulate_clone(flavor)
+        with current_directory(tmp_path):
+            yield ScriptLanguageContainer.create(
+                secrets,
+                name=slc_name,
+                flavor=flavor,
+            )
+
+    return context
+
+
+def test_create(
```

Comment:
Why not using fixture `slc_factory` here?

Reply:
Because the tests also wants to inspect the `git_repo_mock` which is other wise hidden.